File: src/Entity/Provider.php
<?php

namespace App\Entity;

use DateTime;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\EntityManager;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\ProviderRepository")
 */
class Provider
{
    /**
     * @ORM\Id()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $accessGroupName;

    /**
     * @ORM\Column(type="integer")
     */
    private $accessGroupId;

    /**
     * @ORM\Column(type="integer")
     */
    private $totalUses;

    /**
     * @ORM\ManyToMany(targetEntity="App\Entity\Session", inversedBy="providers")
     */
    private $sessions;

    /**
     * @ORM\Column(type="integer")
     */
    private $eventId;

    /**
     * @ORM\Column(type="boolean")
     */
    private $structureDecode;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $name;

    /**
     * @ORM\Column(type="datetime", nullable=true)
     */
    private $modified;

    /**
     * @ORM\Column(type="integer")
     */
    private $basicProductId;

    public function __construct()
    {
        $this->sessions = new ArrayCollection();
    }

    public function create(
        int $id,
        string $accessGroupName,
        int $accessGroupId,
        int $totalUses,
        int $eventId,
        bool $structureDecode,
        string $name,
        int $basicProductId,
        DateTime $modified = null
    )
    {
        $this->id = $id;
        $this->accessGroupName = $accessGroupName;
        $this->accessGroupId = $accessGroupId;
        $this->totalUses = $totalUses;
        $this->eventId = $eventId;
        $this->structureDecode = $structureDecode;
        $this->name = $name;
        $this->basicProductId = $basicProductId;
        $this->modified = $modified;
    }

    public function update(
        string $accessGroupName,
        int $accessGroupId,
        int $totalUses,
        int $eventId,
        bool $structureDecode,
        string $name,
        int $basicProductId,
        DateTime $modified = null
    )
    {
        $this->accessGroupName = $accessGroupName;
        $this->accessGroupId = $accessGroupId;
        $this->totalUses = $totalUses;
        $this->eventId = $eventId;
        $this->structureDecode = $structureDecode;
        $this->name = $name;
        $this->basicProductId = $basicProductId;
        $this->modified = $modified;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId($id): void
    {
        $this->id = $id;
    }

    public function getAccessGroupName(): ?string
    {
        return $this->accessGroupName;
    }

    public function setAccessGroupName(string $accessGroupName): self
    {
        $this->accessGroupName = $accessGroupName;

        return $this;
    }

    public function getAccessGroupId(): ?int
    {
        return $this->accessGroupId;
    }

    public function setAccessGroupId(int $accessGroupId): self
    {
        $this->accessGroupId = $accessGroupId;

        return $this;
    }

    public function getTotalUses(): ?int
    {
        return $this->totalUses;
    }

    public function setTotalUses(int $totalUses): self
    {
        $this->totalUses = $totalUses;

        return $this;
    }

    public function getEventId(): ?int
    {
        return $this->eventId;
    }

    public function setEventId(int $eventId): self
    {
        $this->eventId = $eventId;

        return $this;
    }

    public function getStructureDecode(): ?bool
    {
        return $this->structureDecode;
    }

    public function setStructureDecode(bool $structureDecode): self
    {
        $this->structureDecode = $structureDecode;

        return $this;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;

        return $this;
    }

    public function getBasicProductId(): ?int
    {
        return $this->basicProductId;
    }

    public function setBasicProductId(int $basicProductId): self
    {
        $this->basicProductId = $basicProductId;

        return $this;
    }

    public function getModified(): ?\DateTimeInterface
    {
        return $this->modified;
    }

    public function setModified(?\DateTimeInterface $modified): self
    {
        $this->modified = $modified;

        return $this;
    }

    /**
     * @return Collection|Session[]
     */
    public function getSessions(): Collection
    {
        return $this->sessions;
    }

    public function addSession(Session $session): self
    {
        if (!$this->sessions->contains($session)) {
            $this->sessions[] = $session;
        }

        return $this;
    }

    public function removeSession(Session $session): self
    {
        if ($this->sessions->contains($session)) {
            $this->sessions->removeElement($session);
        }

        return $this;
    }
}

File: src/EasyGOBand/src/main/java/com/assignment/easygoband/vo/Provider.java
package com.assignment.easygoband.vo;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown=true)
public class Provider {

	@JsonProperty("access_group_name")
	private String accessGroupName;

	@JsonProperty("access_group_id")
    private int accessGroupId;

	@JsonProperty("total_uses")
	private int totalUses;

    private List<Session> sessions;

    @JsonProperty("event_id")
    private String eventId;

    @JsonProperty("structure_decode")
    private boolean structureDecode;

    private String name;

    private String modified;

    private String id;

    @JsonProperty("basic_product_id")
    private String basicProductId;

	public String getAccessGroupName() {
		return accessGroupName;
	}

	public void setAccessGroupName(String accessGroupName) {
		this.accessGroupName = accessGroupName;
	}

	public int getAccessGroupId() {
		return accessGroupId;
	}

	public void setAccessGroupId(int accessGroupId) {
		this.accessGroupId = accessGroupId;
	}

	public int getTotalUses() {
		return totalUses;
	}

	public void setTotalUses(int totalUses) {
		this.totalUses = totalUses;
	}

	public List<Session> getSessions() {
		return sessions;
	}

	public void setSessions(List<Session> sessions) {
		this.sessions = sessions;
	}

	public String getEventId() {
		return eventId;
	}

	public void setEventId(String eventId) {
		this.eventId = eventId;
	}

	public boolean isStructureDecode() {
		return structureDecode;
	}

	public void setStructureDecode(boolean structureDecode) {
		this.structureDecode = structureDecode;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getModified() {
		return modified;
	}

	public void setModified(String modified) {
		this.modified = modified;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getBasicProductId() {
		return basicProductId;
	}

	public void setBasicProductId(String basicProductId) {
		this.basicProductId = basicProductId;
	}
    
    
    
}
